src/mz/src/api.rs

```diff
@@ -184,7 +191,13 @@ pub async fn get_region(
         .send()
         .await?;
     ensure!(response.status().is_success());
-    Ok(response.json::<Region>().await?)
+
+    match response.json::<APIResponse>().await? {
+        Ok(APIResponse::Empty) => TODO err emptiness,
```

Comment:
Are you looking for something like this? ([bail!](https://docs.rs/anyhow/latest/anyhow/macro.bail.html))
```suggestion
        Ok(APIResponse::Empty) => bail!("The region is empty"),
```